packages/lesswrong/components/form-components/FormComponentFriendlyDisplayNameInput.tsx

```diff
@@ -0,0 +1,102 @@
+import React, { ReactNode } from "react";
+import { Components, registerComponent } from "../../lib/vulcan-lib";
+
+const styles = (theme: ThemeType) => ({
+  inputRow: {
+    display: "flex",
+    gap: `12px`,
+    [theme.breakpoints.down('xs')]: {
+      gap: `6px`,
+    }
+  },
+  formInput: {
+    flex: "1",
+  },
+  copyChips: {
+    display: "flex",
+    gap: "6px",
+    height: "min-content",
+    marginTop: "auto",
+    marginRight: 6,
+    marginBottom: 8,
+    [theme.breakpoints.down('sm')]: {
+      marginRight: 0,
+    }
+  },
+  blurb: {
+    marginTop: 6,
+    fontWeight: 450,
+    color: theme.palette.grey[600],
+  },
+  link: {
+    textDecoration: "underline",
+    textUnderlineOffset: "4px",
+    '&:hover': {
+      textDecoration: "underline",
+    }
+  },
+});
+
+export const FormComponentFriendlyDisplayNameInput = ({
+  value,
+  classes,
+  ...props
+}: {
+  multiline?: boolean;
+  rows?: number;
+  fullWidth?: boolean;
+  startAdornment?: ReactNode;
+  smallBottomMargin?: boolean;
+  className?: string;
+  classes: ClassesType<typeof styles>;
+} & FormComponentProps<string>) => {
+  const { FormComponentFriendlyTextInput, CopyChip } = Components;
+
+  const blurbContent = (
+    <span>
+      If you've taken the{" "}
+      <a
+        className={classes.link}
+        href="https://www.givingwhatwecan.org/pledge"
+        target="_blank"
+        rel="noopener noreferrer"
+      >
+        🔸10% Pledge
+      </a>{" "}
+      or{" "}
+      <a
+        className={classes.link}
+        href="https://www.givingwhatwecan.org/pledge#pledge-options"
+        target="_blank"
+        rel="noopener noreferrer"
+      >
+        🔹Trial Pledge
+      </a>, consider adding a diamond to your name to show others
+    </span>
+  );
+
+  return (
```

Comment:
A couple of styling things:

<img width="732" alt="Screenshot 2024-07-22 at 10 40 05" src="https://github.com/user-attachments/assets/f9899566-e97c-45a6-ad28-97bbe25bacd6">

- In dark mode the chips are darker than the input which looks a bit odd (looks correct in light mode though)
- Maybe a little more padding between the input and the text underneath?
- Vertical alignment inside the copy chips looks off (maybe a firefox thing?). Maybe they should be the same height as the input too - not sure.

Reply:
I've hopefully fixed all of these, new screenshot:
![Screenshot 2024-07-22 at 11 35 51](https://github.com/user-attachments/assets/a0033c55-012e-48bf-8236-c5e26723c533)

Reply:
> Vertical alignment inside the copy chips looks off (maybe a firefox thing?). Maybe they should be the same height as the input too - not sure

Agree with this! Would make the copy boxes as tall as the input. 

Also think they may be a little complex looking (took me a while to process what was going on first time I saw it)–I'd maybe do a direct **add** instead of copying (which would make the UI more like "Add 🔶" ) and the emoji could be pasted at the end of the text?

Reply:
This is what it looks like with the buttons full height:
![Screenshot 2024-07-22 at 13 40 16](https://github.com/user-attachments/assets/9fc69e58-d92b-4b6a-8429-bc5869accfe0)

And with "Add 🔶", which I think doesn't look great as is, although maybe we could change the design more to make it work (such as not having these buttons at the side at all):
![Screenshot 2024-07-22 at 13 44 32](https://github.com/user-attachments/assets/e55cc128-b255-4409-87ba-def9733fadad)
![Screenshot 2024-07-22 at 13 44 50](https://github.com/user-attachments/assets/fe4e3dd9-0081-4026-87f7-b6886d62a28c)

Personally I prefer copying over directly adding because it means the form will never change the value you input yourself, which matches the expectation from the rest of the form. If it added the diamond for you I (as a user) would be concerned that if I then deleted it it might end up in a weird state